File: src/java/org/elastos/hive/subscription/PricingPlan.java
package org.elastos.hive.subscription;

public class PricingPlan {
	private float amount;
	private String currency;
	private int maxStorage;
	private String name;
	private int serviceDays;

	public PricingPlan setAmount(float amount) {
		this.amount = amount;
		return this;
	}

	public PricingPlan setCurrency(String currency) {
		this.currency = currency;
		return this;
	}

	public PricingPlan setMaxStorage(int maxStorage) {
		this.maxStorage = maxStorage;
		return this;
	}

	public PricingPlan setName(String name) {
		this.name = name;
		return this;
	}

	public PricingPlan setServiceDays(int serviceDays) {
		this.serviceDays = serviceDays;
		return this;
	}

	public float getAmount() {
		return amount;
	}

	public String getCurrency() {
		return currency;
	}

	public int getMaxStorage() {
		return maxStorage;
	}

	public String getName() {
		return name;
	}

	public int getServiceDays() {
		return serviceDays;
	}
}

File: src/service/subscription/pricingplan.ts
/**
 * The pricing plan is for upgrading the service of the vault or the backup.
 */
export class PricingPlan {
	private amount: number;
	private currency: string;
	private maxStorage: number;
	private name: string;
	private serviceDays: number;

	public setAmount(amount: number): PricingPlan {
		this.amount = amount;
		return this;
	}

	public setCurrency(currency: string): PricingPlan {
		this.currency = currency;
		return this;
	}

	public setMaxStorage(maxStorage: number): PricingPlan {
		this.maxStorage = maxStorage;
		return this;
	}

	public setName(name: string): PricingPlan {
		this.name = name;
		return this;
	}

	public setServiceDays(serviceDays: number): PricingPlan {
		this.serviceDays = serviceDays;
		return this;
	}

	public getAmount(): number {
		return this.amount;
	}

	public getCurrency(): string {
		return this.currency;
	}

	public getMaxStorage(): number {
		return this.maxStorage;
	}

	public getName(): string {
		return this.name;
	}

	public getServiceDays(): number {
		return this.serviceDays;
	}
}
